test/suite/test_checkpoint07.py

```diff
@@ -0,0 +1,192 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_checkpoint07.py
+# Test that the checkpoints timing statistics are populated as expected.
+
+import wiredtiger, wttest
+from wiredtiger import stat
+from wtdataset import SimpleDataSet
+
+def timestamp_str(t):
+    return '%x' % t
+
+class test_checkpoint07(wttest.WiredTigerTestCase):
+    conn_config = 'cache_size=50MB,log=(enabled),statistics=(all)'
+    session_config = 'isolation=snapshot'
+
+    def get_stat(self, uri):
+        stat_uri = 'statistics:' + uri
+        stat_cursor = self.session.open_cursor(stat_uri)
+        val = stat_cursor[stat.dsrc.btree_clean_checkpoint_timer][2]
+        stat_cursor.close()
+        return val
+
+    def count_checkpoints(self):
+        metadata_cursor = self.session.open_cursor('metadata:', None, None)
+
+        nckpt = 0
+        while metadata_cursor.next() == 0:
+            key = metadata_cursor.get_key()
+            value = metadata_cursor[key]
+            nckpt = nckpt + value.count("WiredTigerCheckpoint")
+        metadata_cursor.close()
+        return nckpt
+
+    def test_checkpoint07(self):
+        self.uri1 = 'table:ckpt05.1'
+        self.file1 = 'file:ckpt05.1.wt'
+        self.uri2 = 'table:ckpt05.2'
+        self.file2 = 'file:ckpt05.2.wt'
+        self.uri3 = 'table:ckpt05.3'
+        self.file3 = 'file:ckpt05.3.wt'
+        self.session.create(self.uri1, 'key_format=i,value_format=i')
+        self.session.create(self.uri2, 'key_format=i,value_format=i')
+        self.session.create(self.uri3, 'key_format=i,value_format=i')
+
+        # Setup: Insert some data and checkpoint it. Then modify only
+        # the data in the first table and checkpoint. Verify the clean skip
+        # timer is not set for the modified table and is set for the clean one.
+        c1 = self.session.open_cursor(self.uri1, None)
+        c2 = self.session.open_cursor(self.uri2, None)
+        c3 = self.session.open_cursor(self.uri3, None)
+        c1[1] = 1
+        c2[1] = 1
+        c3[1] = 1
+        self.session.checkpoint(None)
+        c1[2] = 2
+        self.session.checkpoint(None)
+        val1 = self.get_stat(self.file1)
+        self.assertEqual(val1, 0)
+        val2 = self.get_stat(self.file2)
+        self.assertNotEqual(val2, 0)
+        val3 = self.get_stat(self.file3)
+        self.assertNotEqual(val3, 0)
+        # It is possible that we could span the second timer when processing table
+        # two and table three during the checkpoint. If they're different check
+        # they are within 1 second of each other.
```

Comment:
The code doesn't seem to match the comment here, "If they're different check they are within 1 second of each other."